internal/manifests/targetallocator/configmap.go

```diff
@@ -52,7 +52,9 @@ func ConfigMap(params manifests.Params) (*corev1.ConfigMap, error) {
 
 	taConfig := make(map[interface{}]interface{})
 	prometheusCRConfig := make(map[interface{}]interface{})
-	taConfig["label_selector"] = manifestutils.SelectorLabels(params.OtelCol.ObjectMeta, collector.ComponentOpenTelemetryCollector)
+	taConfig["collector_selector"] = map[string]any{
+		"matchlabels": manifestutils.SelectorLabels(params.OtelCol.ObjectMeta, collector.ComponentOpenTelemetryCollector),
+	}
```

Comment:
one issue that may arise with this change is that users cannot use the newer operator with an older TA version. This could be problematic in this scenario:
* a company runs a mirrored image of the TA due to a security requirement for the image in the collector CRD
* engineer bumps the operator version which no longer writes the former selector
* TA can no longer startup until they re-mirror and bump the TA image

I think for now it may make sense to just mark the label_selector as deprecated in the operator and will be removed entirely in v1alpha2. It is totally fine, however, to just remove it entirely from the TA.

Reply:
Do we support running different versions of the operator and targetallocator? I know we don't for the collector, even though things usually work anyway.

How would you mark the label_selector as deprecated? From the operator's perspective, it's just a key in an untyped map.

Reply:
I would just mark it in a comment with a TODO linking to an open issue, this could also be a featuregate that starts enabled by default and then we eventually just delete it?

I'm not sure we've discussed what the operators own compat matrix looks like, but the scenario I linked above could break a lot of users because you cannot usually update the TA image and the operator image in the same release. Doing either one without the other would result in a broken state.

Reply:
A problem with this is that target allocator currently unmarshals its configuration in strict mode, so the old version can't deal with the new value either way. The way I see it, there's two possible approaches:

1. Do an explicit version check and apply the right value.
2. Merge a change that makes config unmarshalling non-strict, wait till that's released, and then merge this PR.

I think I like 2 better, WDYT?

Reply:
argh 🤦 I think i like 2 more as well. Lets get that merged in before #2482